src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1129,18 +1135,23 @@ static BluezConnection * BluezCharacteristicGetBluezConnection(BluezGattCharacte
     }
     else
     {
-        options = g_variant_dict_new(aOptions);
-        v       = g_variant_dict_lookup_value(options, "device", G_VARIANT_TYPE_OBJECT_PATH);
-        VerifyOrExit(v != nullptr, ChipLogError(DeviceLayer, "FAIL: No device option in dictionary (%s)", __func__));
+        options       = g_variant_dict_new(aOptions);
+        option_device = g_variant_dict_lookup_value(options, "device", G_VARIANT_TYPE_OBJECT_PATH);
+        VerifyOrExit(option_device != nullptr, ChipLogError(DeviceLayer, "FAIL: No device in options in %s", __func__););
 
-        retval = static_cast<BluezConnection *>(g_hash_table_lookup(apEndpoint->mpConnMap, g_variant_get_string(v, nullptr)));
+        retval = static_cast<BluezConnection *>(
+            g_hash_table_lookup(apEndpoint->mpConnMap, g_variant_get_string(option_device, nullptr)));
```

Comment:
Does this string increase a reference counter of `option_device`? The doc says that the string is valid as long as the variant is, so I'm not sure if this shouldn't be duplicated, but it's not very clear to me :).